src/debtManagers/VariableDebtManager.sol

```diff
@@ -0,0 +1,132 @@
+    uint public lastUpdate;
+
+    constructor(address _amm, address _helper, address _dbr, address _dola) {
+        lastUpdate = block.timestamp;
+        helper = IHelper(_helper);
+        amm = IDbrAMM(_amm);
+        dbr = IDBR(_dbr);
+        dola = IDOLA(_dola);
+    }
+
+    modifier onlyMarket() {
+        if(!dbr.markets(msg.sender)){
+            revert OnlyMarket();
+        } 
+        _;
+    }
+
+    modifier updateDebt() {
+        _burnDbrDeficit();
+        _;
+    }
+
+    error OnlyMarket();
+    
+    //Should be called when switching fixed rate debt to variable debt and when borrowing variable debt
+    function increaseDebt(address user, uint additionalDebt) external onlyMarket updateDebt {
+        address market = msg.sender;
+        totalDebt += additionalDebt;
+        uint additionalDebtShares;
+        if(totalDebt == 0){
+            additionalDebtShares = additionalDebt * MANTISSA; //Minting a high amount of initial debt shares, as debt per share will increase exponentially over the lifetime of the contract
+        } else {
+            additionalDebtShares = additionalDebt * totalDebtShares / totalDebt; //TODO: Consider rounding up in favour of other users
+        }
+        totalDebtShares += additionalDebtShares;
+        marketDebtShares[market] += additionalDebtShares;
+        debtShares[market][user] += additionalDebtShares;
+    }
+
+    //Should be called when switching variable debt with fixed rate debt, when repaying and when a user is liquidated
+    function decreaseDebt(address user, uint amount) external onlyMarket updateDebt returns(uint){
+        address market = msg.sender;
+        uint userDebt = _debt(market, user);
+        if(userDebt <= amount){
+            totalDebtShares -= debtShares[market][user];
+            marketDebtShares[market] -= debtShares[market][user];
+            debtShares[market][user] = 0;
+            totalDebt -= userDebt;
+            return userDebt;
+        } else {
+            uint removedDebtShares = totalDebtShares * amount / totalDebt;
+            totalDebt -= amount;
+            totalDebtShares -= removedDebtShares; //TODO: Make sure this doesn't underflow
+            marketDebtShares[market] -= removedDebtShares; //TODO: Make sure this doesn't underflow
+            debtShares[market][user] -= removedDebtShares; //TODO: Make sure this doesn't underflow
+            return amount;
+        }
+    }
+
+    function buyHook() external {
+        _burnDbrDeficit();
+    }
+
+    function _burnDbrDeficit() internal {
+        if(lastUpdate < block.timestamp){
+            uint _dbrDeficit = dbrDeficit();
+            uint dolaNeeded = helper.dolaNeededForDbr(_dbrDeficit);
+            totalDebt += dolaNeeded;
+            lastUpdate = block.timestamp;
+            dola.mint(address(this), dolaNeeded);
+            amm.burnDbr(dolaNeeded, _dbrDeficit);
+        }
+    }
+
+    function dbrDeficit() public view returns (uint){
```

Comment:
In the market contract on the #56 branch, the interface of this function has an address borrower param. Seems to mismatch here.